static/js/components/GroupSources.jsx

```diff
@@ -0,0 +1,351 @@
+import React, { useEffect, Suspense } from "react";
+import PropTypes from "prop-types";
+import { useSelector, useDispatch } from "react-redux";
+
+import TableCell from "@material-ui/core/TableCell";
+import TableRow from "@material-ui/core/TableRow";
+
+import Typography from "@material-ui/core/Typography";
+import IconButton from "@material-ui/core/IconButton";
+import Grid from "@material-ui/core/Grid";
+import Chip from "@material-ui/core/Chip";
+import Link from "@material-ui/core/Link";
+import PictureAsPdfIcon from "@material-ui/icons/PictureAsPdf";
+
+import MUIDataTable from "mui-datatables";
+import { makeStyles } from "@material-ui/core/styles";
+
+import Tooltip from "@material-ui/core/Tooltip";
+import GroupIcon from "@material-ui/icons/Group";
+
+import dayjs from "dayjs";
+
+import { ra_to_hours, dec_to_hours } from "../units";
+import * as SourcesAction from "../ducks/sources";
+import styles from "./CommentList.css";
+import ThumbnailList from "./ThumbnailList";
+import UserAvatar from "./UserAvatar";
+import ShowClassification from "./ShowClassification";
+
+const VegaPlot = React.lazy(() => import("./VegaPlot"));
+
+const useStyles = makeStyles((theme) => ({
+  chip: {
+    margin: theme.spacing(0.5),
+  },
+  source: {},
+}));
+
+const GroupSources = ({ route }) => {
+  const dispatch = useDispatch();
+  const sources = useSelector((state) => state.sources.groupSources);
+  const groups = useSelector((state) => state.groups.user);
+  const { taxonomyList } = useSelector((state) => state.taxonomies);
+  const classes = useStyles();
+
+  // Color styling
+  const userColorTheme = useSelector(
+    (state) => state.profile.preferences.theme
+  );
+
+  const commentStyle =
+    userColorTheme === "dark" ? styles.commentDark : styles.comment;
+
+  // Load the group sources
+  useEffect(() => {
+    dispatch(SourcesAction.fetchGroupSources({ group_id: route.id }));
+  }, [route.id, dispatch]);
+
+  if (!sources) {
+    return "Loading Sources...";
+  }
+
+  const group_id = parseInt(route.id, 10);
+  let group_name = "";
+
+  if (groups) {
+    const group = groups.find((g) => {
+      // find the object for the group
+      return g.id === group_id;
+    });
+    group_name = group?.name;
+  }
+
+  // This is just passed to MUI datatables options -- not meant to be instantiated directly.
+  const renderPullOutRow = (rowData, rowMeta) => {
+    if (!sources) {
+      return "Loading...";
+    }
```

Comment:
```suggestion
```
duplicate

Reply:
Let's replace it with a spinner?
Import:
`import CircularProgress from "@material-ui/core/CircularProgress";`

```suggestion
    if (!sources) {
      return (
        <div>
          <CircularProgress color="secondary" />
        </div>
      );
    }
```

Reply:
Cool!